### PR TITLE
Feature/#116 refactoring tms connection setup

### PR DESCRIPTION
**Pull request by Harikarthyk**

### CODE REVIEW

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/functions.js [hunk 1]
```diff
@@ -1,3 +1,85 @@
...
+                ns.updateComponentHTML({}, $component);
+            }
+
+            ns.dataReceived = (data, $component) => {
+                // Passing {} because, The values from the model obj are fetched in bellow function definition.
+                ns.updateComponentHTML(data, $component);
+            }
+        }
+
+        init();
```
Comment (wildone) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/functions.js:84]: functions are essentially utils, all init() calls need to be in behaviour.js.

File: tests/content/src/main/content/jcr_root/content/typerefinery-showcase/pages/pages/dashboard/.content.xml [hunk 2]
```diff
@@ -16,16 +16,16 @@
           </container_1_11>
           <container_1_1 jcr:primaryType= "nt:unstructured" sling:resourceType= "/apps/typerefinery/components/structure/container" smColSize= "24" mdColSize= "16" lgColSize= "12">
```
Comment (wildone) [inline tests/content/src/main/content/jcr_root/content/typerefinery-showcase/pages/pages/dashboard/.content.xml:17]: where are the functional tests for this?

**Review by wildone — CHANGES_REQUESTED** [review]

File: application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientLibs/helper.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+const rootEle = document.querySelector(':root');
```
[the remark this answers is not in the document]
Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientLibs/helper.js:1]: I wanted a global variable for the theming.

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,3 +1,3 @@
 {
-    "java.configuration.updateBuildConfiguration": "automatic"
+    "java.configuration.updateBuildConfiguration": "interactive"
```
[the remark this answers is not in the document]
Comment (Harikarthyk) [reply .vscode/settings.json]: That was automatically updated by the vs code. I have reverted this back.

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/functions.js [hunk 5]
```diff
@@ -1,3 +1,85 @@
-//this is the namesapce for the functions for this component
-window.Typerefinery.Components.Widgets.Ticker = {
-}
+//this is the namespace for the functions for this component
+window.Typerefinery.Components.Widgets.Ticker = {}
+    ; (function (ns, window, document) {
+        console.log('INIT Function')
```
[the remark this answers is not in the document]
Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/functions.js]: Added use strict

File: tests/content/src/main/content/jcr_root/content/typerefinery-showcase/pages/pages/dashboard/.content.xml (hunk 2 above)
Comment (Harikarthyk) [reply tests/content/src/main/content/jcr_root/content/typerefinery-showcase/pages/pages/dashboard/.content.xml:17]: I think functional test cases are handled in different PR

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/functions.js (hunk 1 above)
Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/functions.js]: Ok. changed them

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/behaviour.js [hunk 6]
```diff
@@ -1 +1,32 @@
-//this is the namesapce activating this component
\ No newline at end of file
+//this is the namespace for the functions for this component
+
+; (function (document) {
+    function init() {
```
[the remark this answers is not in the document]
Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/behaviour.js:6]: Added use strict

File: application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientLibs/helper.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (wildone) [reply application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientLibs/helper.js:1]: so and a global namespace is not good enought? Rule #1 Nothing goes into a global space apart from Namespaces.

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/behaviour.js [hunk 7]
```diff
@@ -1 +1,34 @@
...
+            // MODEL 
+            else {
+                ns.modelDataConnected($component);
+            }
+        });
+    }
+    
+    $(function () {
+        init();
+    });
```
Comment (wildone) [inline application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/behaviour.js:33]: what is this!? cant this be executed from behaviour, on some document ready event?

File: application/backend/src/main/resources/apps/typerefinery/components/graphs/xAxisBarChart/script.js [hunk 8]
```diff
@@ -38,7 +38,9 @@ function xAxisBarChartComponent(component, id) {
       drawXAxisBarChart(defaultData, id);
```
Comment (wildone) [inline application/backend/src/main/resources/apps/typerefinery/components/graphs/xAxisBarChart/script.js:38]: this file needs tobe refactored into a namespace. what is the meaning of xAxisBarChart? will there be a new component graphs/yAxisBarChart?

**Review by wildone — CHANGES_REQUESTED** [review]
did you add functional tests?

**Comment by wildone** [comment]
@Shivam-apna @Harikarthyk @rohitgupta274509 @Rahul139-stack need to add functional test for all components.

File: application/backend/src/main/resources/apps/typerefinery/components/graphs/xAxisBarChart/script.js (hunk 8 above)
Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/graphs/xAxisBarChart/script.js:38]: Yeah, these things will be refactored.
Scripts will be moved to clientLibs as implemented in the ticker component.

Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/graphs/xAxisBarChart/script.js:38]: In this, PR only the ticker component is refactored.

File: application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/behaviour.js (hunk 7 above)
Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/widgets/ticker/clientlibs/behaviour.js:33]: This is getting triggered in the behaviour.js only and getting started on ready.

This is the short end of the document.onReady in jquery

File: application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientLibs/helper.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Harikarthyk) [reply application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientLibs/helper.js:1]: Ok, in this PR only the ticker component is updated with the namespaces. Can we create separate individual issues for each component and merge them individually?

**Comment by Harikarthyk** [comment]
Closing this PR. Please refer to #118

**Comment by Harikarthyk** [comment]
Closing this PR. Please refer to #118

**Pull request closed without merging**